salt/states/archive.py

```diff
@@ -1239,7 +1259,19 @@ def extracted(name,
             __states__['file.directory'](name, user=user, makedirs=True)
             created_destdir = True
 
-        log.debug('Extracting %s to %s', cached, name)
+        if skip_same_file:
+            origin_name = name
+            tmp_filename = salt.utils.hashutils.md5_digest(os.path.basename(cached))
+            name = os.path.join(os.path.dirname(cached), tmp_filename)
```

Comment:
We should not be changing the value of `name` in a state. It's just not a good idea. The `name` parameter is something fundamental to the state being run. If we need to operate on a different path, then we should be assigning that alternate path to a different variable name, not overwriting `name`.